### ISSUE TITLE
The number of retries to check whether the server is healthy is incorrect.

### ISSUE THREAD

**Issue by czs007** [milvus-io/pymilvus#661]
The number of retries to check whether the server is healthy is incorrect.
Let's take a look at the code：

```
def _wait_for_healthy(self, timeout=30, retry=10):
        with self._connection() as handler:
            start_time = time.time()
            while retry > 0:
                if (time.time() - start_time > timeout):
                    break
                try:
                    status = handler.fake_register_link(timeout)
                    if status.error_code == 0:
                        self._deploy_mode = status.reason
                        return
                except:
                    pass
                finally:
                    time.sleep(1)
                    retry -= 1
            raise Exception("server is not healthy, please try again later") 
```
   
In the line `status = handler.fake_register_link(timeout)`, when the server is unavailable, the `timeout` time will be blocked.
Then return to the while loop again, this line `if (time.time() - start_time > timeout)` of judgment will be True, and then exit the loop.  In total, we only retry once.

suggest to modify like this:
```
    def _wait_for_healthy(self, timeout=3, retry=10):
        with self._connection() as handler:
            while retry > 0:
                is_except = False
                try:
                    status = handler.fake_register_link(timeout)
                    if status.error_code == 0:
                        self._deploy_mode = status.reason
                        return
                except:
                    is_except = True
                finally:
                    sleep_time = 1 if is_except else timeout
                    time.sleep(sleep_time)
                    retry -= 1
            raise Exception("server is not healthy, please try again later")
```

**Comment by RazaKhanGit** [comment]
Can I work on this issue?

**Comment by XuanYang-cn** [comment]
@RazaKhanGit Of course! I'll assign you to this issue, thanks!

/assign @RazaKhanGit

**Comment by XuanYang-cn** [comment]
Closing for already fixed on #730
